scripts/microshift-cleanup-data.sh

```diff
@@ -0,0 +1,102 @@
+#!/bin/bash
+
+SCRIPT_NAME=$(basename $0)
+# The 'openshift-ovn-kubernetes' namespace must be in the end of the list
+# to allow for pod deletion when MicroShift is stopped
+PODS_NS_LIST=(openshift-service-ca openshift-ingress openshift-dns openshift-storage openshift-ovn-kubernetes)
+FULL_CLEAN=true
+
+function usage() {
+    echo "Stop all MicroShift services, also cleaning their data"
+    echo ""
+    echo "Usage: ${SCRIPT_NAME} <--all | --ovn>"
+    echo "   --all   Clean all MicroShift and OVN data"
+    echo "   --ovn   Clean OVN data only"
+    exit 1
+}
+
+# Parse command line
+case $1 in
+--all)
+    ;;
+--ovn)
+    FULL_CLEAN=false
+    ;;
+*)
+    usage
+    ;;
+esac
+
+# Exit if the current user is not 'root'
+if [ $(id -u) -ne 0 ] ; then
+    echo "The '${SCRIPT_NAME}' script must be run with the 'root' user privileges"
+    exit 0
+fi
+
+if ${FULL_CLEAN} ; then
+    echo "DATA LOSS WARNING: Do you wish to stop and clean ALL MicroShift data AND cri-o container workloads?"
+    select yn in "Yes" "No"; do
+        case "${yn}" in
+        Yes) 
+            break
+            ;;
+        *)
+            echo "Aborting cleanup" 
+            exit 0
+            ;;
+        esac
+    done
+fi
+
+echo Stopping MicroShift services
+for service in microshift microshift-etcd ; do
+    systemctl stop --now   ${service} 2>/dev/null
+    systemctl reset-failed ${service} 2>/dev/null
+done
+
+if ${FULL_CLEAN} ; then
+    echo Disabling MicroShift services
+    systemctl disable microshift 2>/dev/null
+fi
+pkill -9 --exact microshift
+pkill -9 --exact microshift-etcd
+
+if ${FULL_CLEAN} ; then
+    echo Removing crio container and image storage
+    crio wipe -f &>/dev/null
+    systemctl restart crio
+else
+    echo Removing MicroShift pods
+    for i in ${!PODS_NS_LIST[@]}; do
+        ns=${PODS_NS_LIST[$i]}    
+
+        retries=5
+        while [ ${retries} -gt 0 ] ; do
+            ocp_pods=$(crictl pods --namespace ${ns} -q)
+            if [ "$(echo ${ocp_pods} | wc -w)" -eq 0 ] ; then
+                break
+            fi
+            crictl rmp -f ${ocp_pods} &>/dev/null
+            retries=$(( retries - 1 ))
+        done
+    done
+fi
+
+echo Killing conmon, pause and OVN processes
+ovs-vsctl del-br br-int 2>/dev/null
```

Comment:
Why don't we care about error messages here?

Reply:
We're performing a clean-up operation, so some utilities report errors/warnings that are confusing to the caller. The idea is to get rid of these devices and processes so that `rm -rf /var/run/ovn` command succeeds. The main goal is to clean the ovn database...

Reply:
I'm worried about debugging a situation where the command fails, the error is discarded, and then some other error happens later because the original action could not be taken. Can we structure the logging with "Starting X action" and "Completed X action" messages around the commands to make clear what errors go with what commands and still keep the stderr output?

Reply:
That's a fair point. I changed to code to list the interface with output redirection. Then, we either execute the delete operation on the bridge or get a message that the bridge does not exist.

Please, review the two example outputs and let me know if this is better now.

```
$ sudo ./scripts/microshift-cleanup-data.sh --ovn
Stopping MicroShift services
Removing MicroShift pods
Deleting the br-int interface
Killing conmon, pause and OVN processes
Removing OVN configuration
Cleanup succeeded
```

```
$ sudo ./scripts/microshift-cleanup-data.sh --ovn
Stopping MicroShift services
Removing MicroShift pods
The br-int bridge interface does not exist
Killing conmon, pause and OVN processes
Removing OVN configuration
Cleanup succeeded
```